src/index/mod.rs

```diff
@@ -50,6 +51,14 @@ impl Index {
         Ok(Self { path, api })
     }
 
+    pub fn run_git_gc(&self) {
+        let cmd = format!("cd {} && gc --auto", self.path.to_str().unwrap());
+        let gc = Command::new("sh").args(&["-c", cmd.as_str()]).output();
```

Comment:
```suggestion
        let gc = Command::new("git").arg("-C").arg(&self.path).args(&["gc", "--auto"]).output();
```

Avoids the use of `sh` and potential panic if the path is non-UTF-8.